### PR TITLE
DOC: behavior when slicing with missing bounds

### LINKED ISSUE #16917: Undocumented feature: partial string slicing
#### Code Sample, a copy-pastable example if possible

```python
>>> index = ['abe', 'adam', 'andrew', 'ben', 'brad', 'cal', 'chad', 'dan']
>>> data = [0] * len(index)
>>> df = pd.DataFrame(index=index, data=data, columns=['col'])
>>> df
        col
abe       0
adam      0
andrew    0
ben       0
brad      0
cal       0
chad      0
dan       0

>>> df.loc['ac':'d']
        col
adam      0
andrew    0
ben       0
brad      0
cal       0
chad      0

```
#### Problem description
Partial string slicing is documented for datetimeindexes but is nowhere to be found for string indexes. The index must be ordered for it to work. I couldn't find a single example of this anywhere online. Is this type of slicing encouraged? Should it be documented?

#### Output of ``pd.show_versions()``

<details>
INSTALLED VERSIONS
------------------
commit: None
python: 3.6.1.final.0
python-bits: 64
OS: Darwin
OS-release: 15.6.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.20.3
pytest: 3.0.7
pip: 9.0.1
setuptools: 35.0.2
Cython: 0.25.2
numpy: 1.13.1
scipy: 0.19.0
xarray: None
IPython: 6.0.0
sphinx: 1.5.5
patsy: 0.4.1
dateutil: 2.6.1
pytz: 2017.2
blosc: None
bottleneck: 1.2.0
tables: 3.4.2
numexpr: 2.6.2
feather: None
matplotlib: 2.0.2
openpyxl: 2.4.7
xlrd: 1.0.0
xlwt: 1.2.0
xlsxwriter: 0.9.6
lxml: 3.7.3
bs4: 4.6.0
html5lib: 0.999999999
sqlalchemy: 1.1.9
pymysql: None
psycopg2: None
jinja2: 2.9.6
s3fs: None
pandas_gbq: None
pandas_datareader: 0.3.0.post
</details>

### PR DESCRIPTION
**Pull request by toobaz**
- [x] closes #16917
 - [ ] tests added / passed
 - [x] passes ``git diff upstream/master --name-only -- '*.py' | flake8 --diff`` (On Windows, ``git diff upstream/master -u -- "*.py" | flake8 --diff`` might work as an alternative.)
 - [ ] whatsnew entry

This documents the behavior discussed in #16917. I don't have time now to produce tests (and I suspect some already exist), so feel free to not close that issue, or to open a new one for tests only.

### CODE REVIEW

**Comment by gfyoung** [comment]
@toobaz : you will need to test this too!

**Comment by toobaz** [comment]
> @toobaz : you will need to test this too!

Are you jocking?! Did you read above?

**Comment by jreback** [comment]
Maybe adding a note box instead would be more clear, showing a situation where this could work and where it doesn't work. 

Putting this in the headline docs is a non-starter.

**Comment by gfyoung** [comment]
> Are you jocking?! Did you read above?

It's been a long day, man.  I did not see that. :smile:

**Comment by gfyoung** [comment]
@toobaz : I would prefer to merge only if there were tests.  As you don't have time to do this, I will go ahead and write tests for it unless @tdpetrou wants to jump in.

**Comment by gfyoung** [comment]
BTW, I should add that the doc changes look pretty reasonable.  As I said before, adding tests (if you have time) would make this merge-able.  Otherwise, we can take care of that.

**Comment by toobaz** [comment]
> BTW, I should add that the doc changes look pretty reasonable.

Cool, thanks. I did put writing those tests (and, first, searching for existing ones) in my pandas TODO list, but frankly speaking it's not very high.

**Comment by gfyoung** [comment]
@toobaz : I presume you have allowed edits by maintainers, so I think what we can do is add the missing tests to your branch if you don't have the time.  Then, we can merge this.  How does that sound?

**Comment by toobaz** [comment]
> @toobaz : I presume you have allowed edits by maintainers, so I think what we can do is add the missing tests to your branch if you don't have the time. Then, we can merge this. How does that sound?

Sure, that would be great!

File: doc/source/indexing.rst
Comment (jreback) [inline doc/source/indexing.rst]: as indicated in my comments 
pull these out to a separate note and put in the label section

File: doc/source/indexing.rst
Comment (jreback) [inline doc/source/indexing.rst]: same

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/indexing.rst
Comment (toobaz) [inline doc/source/indexing.rst]: I moved the main discussion in a box, but if I remove the clarification "if present in the index", then the above is actually false. Should I remove all the sentence?

**Comment by toobaz** [comment]
> what we can do is add the missing tests to your branch

Notice the feature I'm describing is actually already tested at several places, including:
https://github.com/pandas-dev/pandas/blob/master/pandas/tests/indexing/test_floats.py#L387
https://github.com/pandas-dev/pandas/blob/master/pandas/tests/indexing/test_floats.py#L502
https://github.com/pandas-dev/pandas/blob/master/pandas/tests/indexing/test_floats.py#L553
https://github.com/pandas-dev/pandas/blob/master/pandas/tests/indexing/test_floats.py#L593
https://github.com/pandas-dev/pandas/blob/master/pandas/tests/indexing/test_loc.py#L307
with floats, ints and datetimes - but possibly never with strings, which were the main concern in #16917.

File: doc/source/indexing.rst
Comment (jreback) [inline doc/source/indexing.rst]: no that is ok

File: doc/source/indexing.rst
Comment (jreback) [inline doc/source/indexing.rst]: ok, I think we need a mini-example here, this text is pretty dense. can you make a sub-section with a couple of examples. 

This is just not obvious to new users (whom we presume read the documentation). want to make it really clear what is going on. The examples from the issue are pretty good and show the situation.

File: doc/source/indexing.rst
Comment (jreback) [inline doc/source/indexing.rst]: I wouldn't add this reference on this one (maybe in text some where, but not the section head)

File: doc/source/indexing.rst
Comment (jreback) [inline doc/source/indexing.rst]: can you do with with 

```
s = pd.Series(list('abcde'), index=[0, 3, 2, 5, 4])
```

instead. I think a bit more clear on what is getting selected.

File: doc/source/indexing.rst
Comment (jreback) [inline doc/source/indexing.rst]: comma after absent.

File: doc/source/indexing.rst
Comment (jreback) [inline doc/source/indexing.rst]: show

```
s.sort_index()
s.sort_index().loc[1:6]
```

to make this really clear

File: doc/source/indexing.rst
Comment (jreback) [inline doc/source/indexing.rst]: For instance, in the above example,

File: doc/source/indexing.rst
Comment (jreback) [inline doc/source/indexing.rst]: this raises ``KeyError`` not ``IndexError``

File: doc/source/indexing.rst
Comment (jreback) [inline doc/source/indexing.rst]: comma after sorted, you don't need the parens around (and can be compared....)

**Review by jreback — APPROVED** [review]
some nitpicky comments. you can just ping when pushed.

**Comment by jreback** [comment]
ignore pep failure I just fixed

**Comment by toobaz** [comment]
> some nitpicky comments. you can just ping when pushed.

@jreback ping

**Comment by gfyoung** [comment]
I think given the issue you're addressing, an example like what @tdpetrou provided might be useful.

@jreback : I was thinking that we should add tests before merging this?

**Comment by jreback** [comment]
thansks @toobaz 

this is tested in several. Separate issue is whether we want to try show a warning on non-sorted selections.

**Comment by gfyoung** [comment]
> this is tested in several places.

But with partial strings though?  That's what I wasn't sure about.

**Pull request merged**
